crates/lib3h/src/dht/mirror_dht.rs

```diff
@@ -383,67 +391,112 @@ impl MirrorDht {
+            DhtRequestToChild::DropEntryAddress(_) => (),
+
+            DhtRequestToChild::RequestPeer(peer_address) => {
+                let maybe_peer = self.get_peer(&peer_address);
+                let payload = Ok(DhtRequestToChildResponse::RequestPeer(maybe_peer));
+                request.respond(payload);
             }
-        }
+
+            DhtRequestToChild::RequestPeerList => {
+                let list = self.get_peer_list();
+                let payload = Ok(DhtRequestToChildResponse::RequestPeerList(list));
+                request.respond(payload);
+            }
+
+            DhtRequestToChild::RequestThisPeer => {
+                let payload = Ok(DhtRequestToChildResponse::RequestThisPeer(
+                    self.this_peer.clone(),
+                ));
+                request.respond(payload);
+            }
+
+            DhtRequestToChild::RequestEntryAddressList => {
+                let list = self.get_entry_address_list();
+                let payload = Ok(DhtRequestToChildResponse::RequestEntryAddressList(list));
+                request.respond(payload);
+            }
+
+            DhtRequestToChild::RequestAspectsOf(address) => {
+                let maybe_list = self.get_aspects_of(&address);
+                let payload = Ok(DhtRequestToChildResponse::RequestAspectsOf(maybe_list));
+                request.respond(payload);
+            }
+
+            // Ask owner to respond to self
+            DhtRequestToChild::RequestEntry(entry_address) => {
+                self.endpoint_self.request(
+                    DhtContext::RequestEntry(request),
+                    DhtRequestToParent::RequestEntry(entry_address),
+                    Box::new(|_me, context, response| {
+                        let response = {
+                            match response {
+                                GhostCallbackData::Timeout => panic!("timeout"),
+                                GhostCallbackData::Response(response) => match response {
+                                    Err(e) => panic!("{:?}", e),
+                                    Ok(response) => response,
+                                },
+                            }
+                        };
+                        let request = match context {
+                            DhtContext::RequestEntry(request) => request,
+                            _ => panic!("bad context"),
+                        };
+                        if let DhtRequestToParentResponse::RequestEntry(entry_response) = response {
+                            println!("4. In DhtRequestToChild::RequestEntry Responding...");
+                            let payload =
+                                Ok(DhtRequestToChildResponse::RequestEntry(entry_response));
+                            request.respond(payload);
+                        } else {
+                            panic!("bad response to RequestEntry: {:?}", response);
+                        }
```

Comment:
Lots of `panic!`s in this code- is this the type of error handling we want here?

Reply:
Probably not, but I'm guessing we are going to macroize this in some cleaner way anyway so taking the short route here.